### PR TITLE
fix: dynamic properties will be updated before the v-once directive

### PR DESCRIPTION
**Pull request by baiwusanyu-c**
close: #12

### CODE REVIEW

**Comment by autofix-troubleshooter** [comment]
Hi! I'm the <a href="https://autofix.ci">autofix logo<strong>autofix.ci</strong></a> troubleshooter bot.

It looks like you correctly set up a CI job that uses the autofix.ci GitHub Action, but the [**autofix.ci GitHub App**](https://github.com/apps/autofix-ci/) has not been installed for this repository. This means that autofix.ci unfortunately does not have the permissions to fix this pull request. If you are the repository owner, please install the app and then restart the CI workflow! 😃

File: packages/compiler-vapor/__tests__/compile.test.ts [hunk 1]
```diff
@@ -176,6 +176,18 @@ describe('compile', () => {
         expect(code).toMatchSnapshot()
         expect(code).not.contains('effect')
       })
+
+      test('on nested plain element', async () => {
+        const code = await compile(`<div><div :id="foo" v-once /></div>`)
+        expect(code).toMatchSnapshot()
+        expect(code).not.contains('effect')
+      })
```
Comment (LittleSound) [inline packages/compiler-vapor/__tests__/compile.test.ts:184]: This looks very similar to the test I added earlier, which I had previously marked as a failure.

https://github.com/vuejs/core-vapor/blob/877f3fcf96d96efeb51b6442c9f1c56010f5abe8/packages/compiler-vapor/__tests__/compile.test.ts#L174C1-L178C9

Comment (baiwusanyu-c) [reply packages/compiler-vapor/__tests__/compile.test.ts:184]: It should be marked as passed, compared to vue/core. It should not be updated, the test may have to pass as expected

https://play.vuejs.org/#eNp9kU9vwjAMxb+K5Us3wUATN1TQ/ojDdtimbcdcUGq6QJpESdohVf3uc4padkDc7Pd7tp7lFh+dmzU14RLzIL1yEQLF2q2FUZWzPkILnnbQwc7bCjK2ZsIII60JEapQwirxm/tbYfL5aQPPchOpcnobiTuAvFANPEit5GElkMcmE4HrtuWq6/I507NtqYpVxiCD5s4aSeuB5/NxJ04xBs6wU+VsH6zh+G1aIFDayilN/t1FxRkFLqEniW21tr+vvRZ9TdNBlz8kDxf0fTgmTeCHp0C+IYEji1tfUjzhzdcbHbkeYWWLWrP7CvykYHWdMp5sT7UpOPY/X5/2pX+CMuV32BwjmTAclYImZ9f7BfJjnq+cfo67mC36OWE67P4A0pupHQ==

Comment (LittleSound) [reply packages/compiler-vapor/__tests__/compile.test.ts:184]: Maybe you can change it from `test.fails` to `test`.

Comment (baiwusanyu-c) [reply packages/compiler-vapor/__tests__/compile.test.ts:184]: You are right

**Pull request closed without merging**